k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/model/GenericK8sLocatable.kt

```diff
@@ -6,17 +6,16 @@ import com.netflix.spinnaker.keel.api.SimpleLocations
 
 interface GenericK8sLocatable : Locatable<SimpleLocations> {
     val metadata: Map<String, String>
-    val template: K8sObjectManifest
-
+    val template: K8sObjectManifest?
     val namespace: String
-        get() = (template.metadata[NAMESPACE] ?: NAMESPACE_DEFAULT) as String
+        get() = (template?.metadata?.get(NAMESPACE) ?: NAMESPACE_DEFAULT) as String
 
     override val application: String
         get() = metadata.getValue(APPLICATION).toString()
 
     override val id: String
-        get() = "${locations.account}-$namespace-${template.kind}-${template.metadata[NAME]}".toLowerCase()
+        get() = "${locations.account}-$namespace-${template?.kind}-${template?.metadata?.get(NAME)}".toLowerCase()
 
     override val displayName: String
-        get() = "${locations.account}-$namespace-${template.kind}-${template.metadata[NAME]}".toLowerCase()
+        get() = "${locations.account}-$namespace-${template?.kind}-${template?.metadata?.get(NAME)}".toLowerCase()
```

Comment:
if we settle on resource.spec.template.spec then these changes can be reverted? (see the comment on `K8sResourceModel`)

Reply:
I think we can still keep the above and other changes we have in other places in the code, if we keep the template from that last conversation [here](https://github.com/nimakaviani/managed-delivery-k8s-plugin/pull/28#discussion_r624002588). But maybe some validation is good to verify this doesnt blow up.